utils/archive_explorer/tests/test_archive_explorer.py

```diff
@@ -0,0 +1,151 @@
+
+    def test_summary_file_path(self):
+        """ Test path to summary.txt file """
+        self.dac.add_journal_file('GEM', 'test')
+        expected = os.path.join(self.archive_directory, 'NDXGEM', 'Instrument', 'logs',
+                                'journal', 'summary.txt')
+        actual = self.explorer.get_summary_file('GEM')
+        self.assertEqual(actual, expected)
+
+    def test_data_dir_path(self):
+        """ Test path to data directory """
+        expected = os.path.join(self.archive_directory, 'NDXGEM', 'Instrument', 'data')
+        actual = self.explorer.get_data_directory('GEM')
+        self.assertEqual(actual, expected)
+
+    def test_custom_cycle_dir_path(self):
+        """ Test path to custom defined cycle directory """
+        expected = os.path.join(self.archive_directory, 'NDXGEM', 'Instrument', 'data',
+                                'cycle_17_2')
+        actual = self.explorer.get_cycle_directory('GEM', 17, 2)
+        self.assertEqual(actual, expected)
+
+    def test_custom_cycle_dir_does_not_exist(self):
+        """ Test exception raised if custom cycle doesn't exist """
+        self.assertRaises(OSError, self.explorer.get_cycle_directory, "GEM", 10, 1)
+
+    def test_current_cycle_dir_path(self):
+        """ Test path to most recent cycle directory """
+        expected = os.path.join(self.archive_directory, 'NDXGEM', 'Instrument', 'data',
+                                'cycle_18_2')
+        actual = self.explorer.get_current_cycle_directory('GEM')
+        self.assertEqual(actual, expected)
+
+    def test_current_cycle_directory_not_exist(self):
+        """ Test current cycle returns None when directory is empty """
+        self.dac.delete_archive()
+        os.makedirs(os.path.join(self.test_output_directory, 'data-archive', 'NDXGEM',
+                                 'Instrument', 'data'))
+        self.assertIsNone(self.explorer.get_current_cycle_directory('GEM'))
+        shutil.rmtree(os.path.join(self.test_output_directory, 'data-archive'))
+
+    def test_get_most_recent_run_empty(self):
+        """ Test path to most recent run if none exist """
+        self.assertIsNone(self.explorer.get_most_recent_run_since('GEM', datetime.datetime.now()))
+
+    def test_get_most_recent_run_add_before_cut_off(self):
+        """ Test path to most recent run if only run before time cut off exist """
+        self.dac.add_data_to_most_recent_cycle('GEM', ['GEM001.nxs'])
+        test_start_time = datetime.datetime.now()
+        actual = self.explorer.get_most_recent_run_since('GEM', test_start_time)
+        self.assertIsNone(actual)
+
+    def test_get_most_recent_run_single(self):
+        """ Test path valid most recent run """
+        expected = os.path.join(self.archive_directory, 'NDXGEM', 'Instrument', 'data',
+                                'cycle_18_2', 'GEM001.nxs')
+        test_start_time = datetime.datetime.now()
+        time.sleep(0.2)
```

Comment:
Could we comment why this is needed

Reply:
Can remove `time.sleep` in favour of timestamps but the underlying fix for this is too large for this issue. I'll create another